src/utils/createEndpoint.js

```diff
@@ -0,0 +1,58 @@
+import { HTTPSnippet } from 'httpsnippet';
+
+const supportedClients = {
+  curl: {
+    targetId: 'shell',
+    options: { short: true, indent: ' ', prettifyJson: true },
+  },
+};
+
+function validateData(data) {
+  if (typeof data !== 'object') {
+    throw new Error('A data object is required');
+  }
+
+  const requiredFields = [ 'method', 'path'];
+  requiredFields.forEach(field => {
+    if (!(field in data)) {
+      throw new Error(`Request is missing required field ${field}`);
+    }
+  });
+}
+
+function createSnippet(data) {
+  validateData(data);
+  const harObject = {
+    method: data.method,
+    url: new URL(data.path, 'https://service.centrapay.com/').href,
+    httpVersion: 'HTTP/2.0',
+  };
+  if (data.request?.headers) {
+    harObject.headers = Object.entries(data.request.headers).map(([name, value]) => ({ name, value }));
+  }
+  if (data.request?.payload) {
+    harObject.postData = {
+      mimeType: 'application/json',
+      text: JSON.stringify(data.request.payload)
+    };
+  }
+  return new HTTPSnippet(harObject);
+}
+
+export default function createEndpoint(data) {
+  const snippet = createSnippet(data);
```

Comment:
@MeganSteenkamp Suggestion: Call this `createSnippets` (plural). It would allow you to keep the snippet logic in one place and it would keep `createEndpoint` cleaner for future extensions

```js
const requests = createSnippets(data)
```